util/topgen.py

```diff
@@ -268,6 +268,11 @@ def generate_ipgen(top: ConfigT, module: ConfigT, params: ParamsT,
     ipgen_render(module["template_type"], topname, params, out_path)
 
 
+def get_ipgen_params(module: ConfigT) -> ParamsT:
```

Comment:
The issue is with these params: I assume you will declare them in a way similar to the regular ip hjson `param_list`, where each parameter has a number of attributes.

It is also unclear what is the usage or rationale for these parameters. I assume they are a way to change the internals of an IP beyond what regular SV generates can do. Should we expect these parameters to be strictly local or could they be used to modify the top and perhaps other ips? This is why I suggested adding some documentation for them in the ipgen/README.md file.